measureme/src/file_serialization_sink.rs

```diff
@@ -16,7 +23,12 @@ impl SerializationSink for FileSerializationSink {
         let file = fs::File::create(path)?;
 
         Ok(FileSerializationSink {
-            data: Mutex::new((BufWriter::new(file), 0)),
```

Comment:
what is the gain by removing the BufWriter? feels like the new code is similar to the BufWriter code
so will BufWriter::with_capacity(1024*512, file)  give the same result?

Reply:
The difference is that `BufWriter` does not allow directly writing to its buffer, so we are basically re-implementing `BufWriter` here. The interface of `write_atomic` requires there to be a writable output buffer.